milvus/client/check.py

```diff
@@ -50,36 +51,44 @@ def is_legal_array(array):
     return True
 
 
-def is_legal_bin_array(array):
+def is_legal_bin_vector(array):
     if not array or \
             not isinstance(array, bytes) or \
-            len(array) <= 0:
+            len(array) == 0:
         return False
 
     return True
 
 
+def is_legal_numpy_array(array):
+    return False if array is None or array.size == 0 else True
+
+
 def is_legal_records(value):
-    if not isinstance(value, list) or len(value) <= 0:
-        raise ParamError('A vector must be a non-empty, 2-dimensional array and '
-                         'must contain only elements with the float data type.')
+    param_error = ParamError('A vector must be a non-empty, 2-dimensional array and '
```

Comment:
What about binary vectors? Should update this error message.

Reply:
```python
param_error = ParamError('A vector must be a non-empty, 2-dimensional array and '
 'must contain only elements either with the float data type or the bytes data type.')
```

Reply:
solved.